extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java

```diff
@@ -788,6 +792,21 @@ public void transform(TransformationContext context) {
                                 && clazz.declaredAnnotation(ResteasyReactiveDotNames.TYPED) == null) {
                             // Add @Typed(MyResource.class)
                             context.transform().add(createTypedAnnotationInstance(clazz, beanArchiveIndexBuildItem)).done();
+                            return;
+                        }
+                        // check if the class is a bean param
+                        if (beanParams.contains(clazz.name().toString())
+                                && clazz.declaredAnnotation(ResteasyReactiveDotNames.TYPED) == null) {
+                            // Stef: we could use createTypedAnnotationInstance which adds all interfaces but I don't think
```

Comment:
I would use it for the sake of consistency and future debugging :)

Reply:
But why would we want to produce the interfaces as beans? Won't that lead to potential ambiguous issues like the one we want to fix? Also, bean params are only ever injected as their declared type by RESTEasy Reactive: never as interfaces. I understand the consistency issue you raise, but I felt this would introduce broadness that could lead into future bugs, and would rather have restricted this fix to what we know we absolutely require, and add the interfaces later if and when there's a use-case for them.

Reply:
Originally, the impl I added didn't have interfaces and kept just the impl class and that worked fine for RR and for our tests.
But the issue that made me add it (https://github.com/quarkusio/quarkus/issues/29128) shows that user can leverage these classes as beans in various ways that we didn't anticipate. I would generally say that fiddling with types of beans without user knowledge is risky as you might break their injection logic. So the more types you can keep there, the better. It's not like we need to do some extra legwork to make it work either.

Reply:
OK, whatever. Lemme try.